cpu/idle/idle-power-test.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+
+FILE=$(readlink -f ${BASH_SOURCE})
+NAME=$(basename $FILE)
+CDIR=$(dirname $FILE)
+BUSY="$CDIR/busy.sh"
+STRESSPIDS=
+
+start_load_test()
+{
+    cores=$(nproc)
+    count=0
+
+    [ ! -e "$BUSY" ] && echo "missing $BUSY" && exit 1
+
+    for ((i=0; i < $cores; ++i)); do
+        # if hyperthreading is enabled - only start on the even cores
+        [ "$even_only" = true ] && [ $((i % 2)) -eq 1 ] && continue
+        bash $BUSY $i &>/dev/null &
+        STRESSPIDS="$STRESSPIDS ""$!"
+        count=$((count + 1))
+    done
+
+    if [ "$count" -ne "$cores" ]; then
+        echo "count: $count is not equal to target: $1"
+        exit 1
+    fi
+
+    # make sure the processes really starts
+    sleep 5
+
+    for pid in $STRESSPIDS; do
+        result=$(ps -p $pid -o comm=)
+        size=${#result}
+
+        if [ $size -eq 0 ]; then
+            echo "start_load_test: $BUSY is not running!"
+            exit 1
+        fi
+    done
+
+    echo "load processes started"
+}
+
+end_load_test()
+{
+    if [ ! -z "$STRESSPIDS" ]; then
+        for pid in $STRESSPIDS; do
+            kill $pid
+            wait $pid 2>/dev/null
+        done
+        STRESSPIDS=
+        echo "load processes killed"
+    fi
+}
+
+finish()
+{
+    end_load_test
+}
+
+powerunit=0
+get_power_unit()
+{
+    powerunit=$(rdmsr 0x606)
+}
+
+esu=0
+get_energy_status_units()
+{
+    get_power_unit
+    esu=$((16#$powerunit))
+    esu=$((esu >> 8))
+    mask=$((16#1f))
+    esu=$((esu & mask))
+    esu=$((2 ** esu))
+    esu=$(echo "scale=6; 1/$esu" | bc -l | awk '{printf "%f", $0}')
+    echo "energy status units = $esu joules"
+}
+
+get_pkg_energy()
+{
+    reg=$(rdmsr 0x611)
+    reg=$((16#$reg))
+    mask=$((16#ffffffff))
+    reg=$((reg & mask))
+    energy=$(echo "$reg*$esu" | bc -l)
+    echo $energy
+}
+
+average=
+monitor_power()
+{
+    duration=60 # seconds
+
+    last=$(get_pkg_energy)
+    last_time=$(date +%s%N | cut -b1-13)
+    sleep $duration
+    curr=$(get_pkg_energy)
+    curr_time=$(date +%s%N | cut -b1-13)
+    milliseconds=$((curr_time - last_time))
+    joules=$(echo $curr-$last | bc -l)
+    watts=$(echo "$joules*1000/$milliseconds" | bc -l)
+    printf "average power use = %5.2f watts over last %d seconds\n" $watts $duration
+    average=$(echo $watts+0.5 | bc -l)
+    average=${average%.*}
+}
+
+which rdmsr &> /dev/null
+[ $? -ne 0 ] && echo "install msr-utils" && exit 1
```

Comment:
Unless I'm missing something, since the `idle-power-test.sh` is called as part of the `rlRun` command, this will report a failure to developers when it's really a setup issue. Same with a few other lines in here with `exit`

Reply:
Hi @veruu, I removed those 4 lines to check `which rdmsr` and `rdmsr 0x606` from script `idle-power-test.sh`. Now `busy.sh` and `idle-power-test.sh` are moved to subdirectory `utils` as they are just copied from the src url in ticket#923.